src/evolution_metrics.jl

```diff
@@ -0,0 +1,151 @@
+struct IndicatorEvolutionResults{T<:AbstractFloat}
+    t_indicator::AbstractVector{T}
+    x_indicator::AbstractVector{T}
+    t_evolution::AbstractVector{T}
+    x_evolution::AbstractVector{T}
+    surr_x_evolution::AbstractMatrix{T}
+end
+
+"""
+    indicator_evolution(t, x, indicator, n_surr, surr_method, evolution_metric,
+                        wv_indicator_width, wv_indicator_stride,
+                        wv_evolution_width, wv_evolution_stride)
```

Comment:
Okay, here almost everything should be a keyword argument. Only `t, x, indicator` should be arguments, and everything else a keyword with sensible defaults.

Reply:
Additionally, `t` should be an _optional_ argument. If not given, `1:length(x)` is taken as the time vector.

Reply:
solved by introducing the parameters struct